utils/location.py

```diff
@@ -93,27 +86,14 @@ def fetch_new_locations(queries):
                 f"[{i + 1}/{len(new_location_names)}] Unable to geocode institution {inst}"
             )
 
-    return new_location_data
-
-
-def get_mongo_ids():
-    """Return a mapping of every location name to its Mongo ID
-
-    Every entry in the MongoDB "Location" collection has a location
-    data ID. We use these IDs to represent relationships from
-    Articles to their corresponding location data.
-
-    This function returns a dictionary with institution names as
-    keys, and MongoDB IDs as values.
-    """
+    # get existing locations
+    locations = db.Location.objects(institution__in=queries).only("id", "institution")
+    # add new locations
+    if len(new_location_data):
+        print("Inserting new locations into database...")
+        locations += db.create(db.Location, new_location_data)
```

Comment:
Can we do a batch insert here?

Reply:
`db.create` smart_inserts all the new location data at once
I might be misunderstanding what you mean by batch insert